src/hooks/useBlocksInfo.ts

```diff
@@ -26,41 +27,36 @@ interface Props {
 }
 
 const useBlocksInfo = ({ isApiReady, api, chain }: Props) => {
-  const [bestBlock, setBestBlock] = useState('');
-  const [bestBlockFinalized, setBestBlockFinalized] = useState('');
+  const [bestBlock, setBestBlock] = useMountedState('');
+  const [bestBlockFinalized, setBestBlockFinalized] = useMountedState('');
 
   useEffect(() => {
-    if (!api || !isApiReady || !chain) {
-      setBestBlock('');
-      setBestBlockFinalized('');
-      return;
-    }
-
-    let unsubscribeBestNumber: () => void;
-    api.derive.chain
-      .bestNumber((res) => {
-        setBestBlock(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumber = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
-
-    let unsubscribeBestNumberFinalized: () => void;
-    api.derive.chain
-      .bestNumberFinalized((res) => {
-        setBestBlockFinalized(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumberFinalized = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
+    let unsubscribeBestNumber: Promise<VoidFn>;
+    let unsubscribeBestNumberFinalized: Promise<VoidFn>;
+    if (isApiReady && chain) {
+      try {
+        unsubscribeBestNumber = api.derive.chain.bestNumber((res) => {
+          setBestBlock(res.toString());
+        });
 
+        unsubscribeBestNumberFinalized = api.derive.chain.bestNumberFinalized((res) => {
+          setBestBlockFinalized(res.toString());
+        });
+      } catch (e) {
+        logger.error('error reading blocks', e);
+      }
+    }
     return function cleanup() {
-      unsubscribeBestNumberFinalized && unsubscribeBestNumberFinalized();
-      unsubscribeBestNumber && unsubscribeBestNumber();
+      unsubscribeBestNumber
+        .then((u) => {
+          u();
+        })
+        .catch((e) => logger.error('error unsubscribing bestNumber', e));
+      unsubscribeBestNumberFinalized
+        .then((u) => u())
+        .catch((e) => logger.error('error unsubscribing bestNumberFinalized', e));
```

Comment:
I think with `async/await` it would look a bit nicer.

Reply:
let's keep the discussion here: https://github.com/paritytech/parity-bridges-ui/pull/179/files#r641805104